### PR TITLE
React UI: Player updates

### PR DESCRIPTION
**Pull request by bsekachev**
- Moving, zooming, additional canvas handler (zoom handler to select a ROI) 
- Shape activating & changing
- Saving of annotations on the server
- Drawing for shapes/tracks
- Minor fixes in CVAT-Canvas & CVAT-Core
- Absolute import in typescript UI instead of a lot of `../../../../../` (also plugins for eslint & webpack)
- Merge, group, split

So, right side panel is ready except of tags creation
Screenshot from 2020-01-16 17-03-05

### CODE REVIEW

File: cvat-canvas/src/typescript/canvasModel.ts [hunk 1]
```diff
@@ -250,15 +258,41 @@ export class CanvasModelImpl extends MasterImpl implements CanvasModel {
...
         ));
 
-        this.notify(UpdateReasons.FIT_CANVAS);
-        this.notify(UpdateReasons.OBJECTS);
+        this.notify(UpdateReasons.FITTED_CANVAS);
+        this.notify(UpdateReasons.OBJECTS_UPDATED);
+    }
+
+    public dragCanvas(enable: boolean): void {
+        if (enable && this.data.mode !== Mode.IDLE) {
```
Comment (mmarkelov) [inline cvat-canvas/src/typescript/canvasModel.ts]: What about moving this condition in separate function? Like so:
```javascript
public checkIfCanvasIsBusy(enable: boolean): void {
        if (enable && this.data.mode !== Mode.IDLE) {
            throw Error(`Canvas is busy. Action: ${this.data.mode}`);
        }
}
```

File: cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/draw-shape-popover-content.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        } = this.state;
+
+        const {
+            shapeType,
+            labels,
+            onDrawStart,
+            canvasInstance,
+        } = this.props;
+
+        let minimumPoints = 0;
```
Comment (mmarkelov) [inline cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/draw-shape-popover-content.tsx]: Maybe me move this condition to function out of render method? Like so:
```javascript
public getMinimumPoints(shapeType): number {
        let minimumPoints = 0;
        if (shapeType === ShapeType.POINTS) {
            minimumPoints = 1;
        } else if (shapeType === ShapeType.POLYLINE) {
            minimumPoints = 2;
        } else if (shapeType === ShapeType. POLYGON) {
            minimumPoints = 3;
        }
        return minimumPoints;
}

```

File: cvat-ui/src/reducers/annotation-reducer.ts [hunk 3]
```diff
@@ -26,13 +39,19 @@ export default (state = defaultState, action: AnyAction): AnnotationState => {
             };
         }
         case AnnotationActionTypes.GET_JOB_SUCCESS: {
+            const { jobInstance } = action.payload;
```
Comment (mmarkelov) [inline cvat-ui/src/reducers/annotation-reducer.ts]: Maybe add other props here?
```javascript
  const { jobInstance, frame, frameData,  annotations } = action.payload;
```

Comment (bsekachev) [reply cvat-ui/src/reducers/annotation-reducer.ts]: Yes, great idea

File: cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/draw-shape-popover-content.tsx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+                                    step={1}
+                                />
+                            </Col>
+                        </Row>
+                    )
+                }
+                <Row type='flex' justify='space-around'>
+                    <Col span={12}>
+                        <Button
+                            onClick={(): void => {
```
[the remark this answers is not in the document]
Comment (bsekachev) [reply cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/draw-shape-popover-content.tsx]: Yes, we need some refactoring here

File: cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/draw-shape-popover-content.tsx (hunk 2 above)
Comment (bsekachev) [reply cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/draw-shape-popover-content.tsx]: Yes

File: cvat-canvas/src/typescript/canvasModel.ts (hunk 1 above)
Comment (bsekachev) [reply cvat-canvas/src/typescript/canvasModel.ts]: I am not sure it makes sense. 
Yes, this condition is repeated twice, but other methods of ``cvat-canvas`` also check if canvas is busy, but using other conditions.

**Pull request merged**